frame/composable-support/src/abstractions/utils/mod.rs

```diff
@@ -0,0 +1,37 @@
+pub mod decrement;
+pub mod increment;
+pub mod start_at;
+
+/// Helper macro to create a type that can be used in [`DecrementToMax`].
+///
+/// # Usage
+///
+/// ```rust,ignore
+/// error_to_pallet_error!(
+///     SomeErrorType -> PalletErrorVariant;
+/// );
+/// ```
+///
+/// Note that this assumes that the pallet's `Error` and `Config` types are in scope and not
+/// renamed.
+#[macro_export]
+macro_rules! error_to_pallet_error {
+	($($name:ident -> $to:ident;)+) => {
+		$(
+			#[derive(::core::fmt::Debug, ::core::default::Default, ::core::cmp::PartialEq)]
+			pub struct $name;
```

Comment:
pub struct $name(()) ?

and put all errors into mod.

not this pr. i am kind of dislike this macro)

Reply:
It's needed for the IncrementToMax incrementor, since enum variants aren't standalone types.

> and put all errors into mod.

What do you mean here?

Reply:
see error disucussion